codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpProtocolGenerator.kt

```diff
@@ -548,18 +552,56 @@ private class ServerHttpProtocolImplGenerator(
     private fun serverRenderBindingParser(
         binding: HttpBindingDescriptor,
         operationShape: OperationShape,
+        httpBindingGenerator: ServerRequestBindingGenerator,
+        structuredDataParser: StructuredDataParserGenerator,
     ): Writable? {
-        val operationName = symbolProvider.toSymbol(operationShape).name
-        return when (val location = binding.location) {
+        val errorSymbol = if (model.expectShape(binding.member.target) is StringShape) {
+            CargoDependency.SmithyHttpServer(runtimeConfig).asType().member("rejection").member("SmithyRejection")
+        } else {
+            CargoDependency.smithyJson(runtimeConfig).asType().member("deserialize").member("Error")
+        }
+        return when (binding.location) {
             HttpLocation.HEADER -> writable { serverRenderHeaderParser(this, binding, operationShape) }
-            HttpLocation.LABEL -> {
-                null
+            HttpLocation.PAYLOAD -> {
+                val docShapeHandler: RustWriter.(String) -> Unit = { body ->
+                    rust(
+                        "#T($body)",
+                        structuredDataParser.documentParser(operationShape),
+                    )
+                }
+                val structureShapeHandler: RustWriter.(String) -> Unit = { body ->
+                    rust("#T($body)", structuredDataParser.payloadParser(binding.member))
+                }
```

Comment:
Out of curiosity, do you really need to explicitly set the types here? We don't really do it anywhere..

Reply:
You mean the type of the function `RustWriter.(String) -> Unit`? Type inference fails if I omit the signature.

Reply:
Yep, that's what I meant. Thanks for thee clarification.